### PR TITLE
Added support for Handlebars template library

### PR DESCRIPTION
**Pull request by KennethScott**
### What problem does this PR address?
Existing template capabilities are limited.  Enabling use of a library such as Handlebars gives the user much more power and flexibility over the rendered markup.
  
### Is there a related Issue?
N/A
  
Please take a look at my proposed change to allow Handlebars templates.  I think it's a pretty simple change that would greatly enhance the extension.   Please note that I do not have a way to test this, so while I tried to be as careful as possible with the code, it's ultimately more of a proposal for changes than a guarantee of the working change.

Because I tabbed everything over when wrapped the guts of the processTemplate in an IF statement, the diff looks pretty ugly - but in a nutshell, all I really did was the following:
1) added the npm handlebars package
2) modified the getTemplate function to detect a handlebar template by checking for "{{".  If one is found, the template in its entirety is returned (not processed into an array).
3) modified the processTemplate function to detect a handlebar template by checking for an array (the handlebars templates will be a single string).  I then wrapped the guts of the method in an IF statement so that if the handlebar template is not found, processing continues as expected. If one is found, it compiles the template and passes the relevant objects in on the execution.

Please check it out and let me know what you think.  I gave you access to the fork.

Thanks for considering this!

### CODE REVIEW

File: Extensions/XplatGenerateReleaseNotes/V2/ReleaseNotesFunctions.ts [hunk 1]
```diff
@@ -175,246 +185,262 @@ export function processTemplate(template, workItems: WorkItem[], commits: Change
...
+                                                if ((wi.fields["System.Tags"] !== undefined) &&
+                                                    (wi.fields["System.Tags"].toUpperCase() === wiFilter.tags.join("; ").toUpperCase())) {
+                                                    agentApi.logDebug (`${addSpace(modeStack.length + 4)} Tags match, need to check fields`);
+                                                    okToAdd = true;
+                                                } else {
+                                                    agentApi.logDebug (`${addSpace(modeStack.length + 4)} Tags do not match, no need to check fields`);
+                                                    okToAdd = false;
+                                                }
                                             } else {
-                                                agentApi.logDebug (`${addSpace(modeStack.length + 4)} Tags do not match, no need to check fields`);
```
Comment (rfennell) [inline Extensions/XplatGenerateReleaseNotes/V2/ReleaseNotesFunctions.ts]: This changes behaviour for the existing code path, empty tags did not match previously now it does

**Comment by rfennell** [comment]
Thanks for the PR, interesting idea.

I can see that is going to take a while to have a look at. I can see for a start that the CI build has failed, so I need to check that.

Do you have a sample handbars template can add to my release pipeline as a integration test?

**Comment by rfennell** [comment]
Ok I have this branch running in a test harness. It basically works. I used the template

```
## Notes for release  {{releaseDetails.releaseDefinition.name}}    
**Release Number**  : {{releaseDetails.name}}
**Release completed** : {{releaseDetails.modifiedOn}}     
**Build Number**: {{buildDetails.id}}
{#if buildDetails.triggerInfo['pr.number']}
**Build Trigger PR Number**: {buildDetails.triggerInfo['pr.number']
{/if}
**Compared Release Number**  : {{compareReleaseDetails.name}}    

### Associated Work Items ({{workItems.length}})

{{#each workItems}}
*  **{{this.id}}** {{this.fields['System.Title']}}
   - **WIT** {{this.fields['System.WorkItemType']}} 
   - **Tags** {{this.fields['System.Tags']}}
{{/each}}

### Associated commits ({{commits.length}})
{{#each commits}}
* ** ID{{this.id}}** 
   -  **Message:** {{this.message}}
   -  **Commited by:** {{this.author.displayName}} 
{{/each}}

```

Most of it works, but I am being let down by my handlebars knowledge. The construct `{{this.fields['System.Title']}}` fail with the error

```
2020-03-08T14:39:42.5438959Z ##[error]Error: Parse error on line 13:
...{this.id}}** {{this.fields['System.Title
-----------------------^
Expecting 'ID', got 'INVALID'
```

I think the `lookup` helper is needed

**Comment by rfennell** [comment]
@KennethScott , I think I have it working now, the template needs to be

```
## Notes for release  {{releaseDetails.releaseDefinition.name}}    
**Release Number**  : {{releaseDetails.name}}
**Release completed** : {{releaseDetails.modifiedOn}}     
**Build Number**: {{buildDetails.id}}
**Compared Release Number**  : {{compareReleaseDetails.name}}    

### Associated Work Items ({{workItems.length}})

{{#each workItems}}
*  **{{this.id}}**  {{lookup this.fields 'System.Title'}}
   - **WIT** {{lookup this.fields 'System.WorkItemType'}} 
   - **Tags** {{lookup this.fields 'System.Tags'}}
{{/each}}

### Associated commits ({{commits.length}})
{{#each commits}}
* ** ID{{this.id}}** 
   -  **Message:** {{this.message}}
   -  **Commited by:** {{this.author.displayName}} 
{{/each}}
```
In addition to your changes I have updated added a sample template and updated the documentation readme 

This is going to be released as 2.27.x. If you let me know your Azure DevOps organisation I can grant you access to you so you can give the private test release a try before I make it public. But it does seem to be passing all my old acceptance tests and the new one I have created for Handlebars

File: Extensions/XplatGenerateReleaseNotes/V2/ReleaseNotesFunctions.ts (hunk 1 above)
Comment (rfennell) [reply Extensions/XplatGenerateReleaseNotes/V2/ReleaseNotesFunctions.ts]: I checked this was OK

**Comment by KennethScott** [comment]
That is fantastic news!  Great job getting it incorporated so quickly!  I am embarrassed to say I don't know what our devops organization name is off-hand, but I've emailed a couple of guys that should know.  I'm assuming it's our company name, but I want to know for sure before I tell you anything.  It may be tomorrow when I'm back in the office before I can really try it out anyway, but I'll let you know ASAP.  The template looks just like I would expect, so I'm sure it's fine.  Thanks again for the quick response - I think this is a really powerful enhancement to the extension!

**Comment by rfennell** [comment]
If you are not the owner of your organisation (by the way it is https://dev.azure.com/<your org>) I think I will just release this version. It has passed by tests without issue.

This means you will be able to use the new template straightaway as 2.27.4.

I will complete this PR, if there are problems create a new issue

**Pull request merged**